File: reports/Радиончик Светлана Сергеевна/Lab 6/src/Task 2/Decorator.cs
namespace Lab6_2
{
	class Decorator : Level
	{
		protected Level _level;

		public Decorator (Level level)
		{
			this._level = level;
		}

		public void SetLevel(Level level)
		{
			this._level = level;
		}

		public override string GetLevel()
		{
			if (this._level != null)
			{
				return this._level.GetLevel();
			}
			else
			{
				return string.Empty;
			}
		}

		public override string AddToFavourites()
		{
			if (this._level != null)
			{
				return this._level.AddToFavourites();
			}
			else
			{
				return string.Empty;
			}
		}

		public override string GetDiscont()
		{
			if (this._level != null)
			{
				return this._level.GetDiscont();
			}
			else
			{
				return string.Empty;
			}
		}
	}
}

File: Reports/lab6/src/2/Decorator.java
class Decorator implements Level {

    protected Level mLevel;
    protected String mLevelName;

    public Decorator(Level level) {
        this.mLevel = level;
    }

    @Override
    public String getLevel() {
        return this.mLevel.getLevel();
    }

    @Override
    public String buyBook(String name) {
        return this.mLevel.buyBook(name);
    }

    @Override
    public String addToFavourites(String name) {
        return this.mLevel.addToFavourites(name);
    }

    @Override
    public String leaveMark(String name, String mark) {
        return this.mLevel.leaveMark(name, mark);
    }
}